lib/plenario_aot/aot_actions.ex

```diff
@@ -0,0 +1,153 @@
+defmodule PlenarioAot.AotActions do
+  require Logger
+
+  import Ecto.Query
+
+  alias Plenario.Repo
+
+  alias PlenarioAot.{AotMeta, AotData, AotObservation}
+
+  def create_meta(network_name, source_url) do
+    params = %{network_name: network_name, source_url: source_url}
+    meta = %AotMeta{}
+
+    Logger.info("Creating new AoT Meta with params `#{inspect(params)}`")
+
+    AotMeta.changeset(meta, params)
+    |> Repo.insert()
+  end
+
+  def list_metas, do: Repo.all(AotMeta)
+
+  def get_meta(identifier) when is_integer(identifier), do: Repo.get_by(AotMeta, id: identifier)
+  def get_meta(identifier) when is_bitstring(identifier), do: Repo.get_by(AotMeta, slug: identifier)
+
+  def update_meta(%AotMeta{} = meta, params \\ []) do
+    params = Enum.into(params, %{})
+
+    Logger.info("Updating AoT Meta `#{meta.network_name}` with #{inspect(params)}")
+
+    AotMeta.changeset(meta, params)
+    |> Repo.update()
+  end
+
+  def compute_and_update_meta_bbox(%AotMeta{} = meta) do
```

Comment:
Just do it in one shot... it makes it easier to read when you're looking at the scheduled job function.